From 3f6f4964dc79ae986f44afe1687922381f237edd Mon Sep 17 00:00:00 2001
From: Peter Kjellerstedt <pkj@axis.com>
Date: Wed, 3 Mar 2021 12:47:28 +0100
Subject: [PATCH] minstall: Correctly set uid/gid of installed files

In commit caab4d3d, the uid and gid arguments passed to os.chown() by
set_chown() were accidentally swapped, causing files to end up with
incorrect owner/group if the owner and group are not the same.

Upstream-Status: Backport [https://github.com/mesonbuild/meson/commit/6226ac26ef63335bfb817db02b3f295c78214a82]
Signed-off-by: Peter Kjellerstedt <peter.kjellerstedt@axis.com>
---
 mesonbuild/minstall.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/minstall.py b/mesonbuild/minstall.py
index 785ff5869..07da408aa 100644
--- a/mesonbuild/minstall.py
+++ b/mesonbuild/minstall.py
@@ -148,7 +148,7 @@ def set_chown(path: str, user: T.Optional[str] = None, group: T.Optional[str] =
         Use a real function rather than a lambda to help mypy out. Also real
         functions are faster.
         """
-        real_os_chown(path, gid, uid, dir_fd=dir_fd, follow_symlinks=follow_symlinks)
+        real_os_chown(path, uid, gid, dir_fd=dir_fd, follow_symlinks=follow_symlinks)
 
     try:
         os.chown = chown
